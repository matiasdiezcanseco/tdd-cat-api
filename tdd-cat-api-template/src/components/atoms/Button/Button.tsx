import Loading from '../Loading/Loading'

interface IButton {
  text: string
  onClick: () => void
  loading?: boolean
}

const Button: React.FC<IButton> = ({}) => {
  return (
    <button className="min-w-[80px] flex justify-center p-2 text-gray-200 rounded-md bg-slate-600"></button>
  )
}

export default Button
