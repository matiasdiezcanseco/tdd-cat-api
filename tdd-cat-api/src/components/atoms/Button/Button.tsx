import Loading from '../Loading/Loading'

interface IButton {
  text: string
  onClick: () => void
  loading?: boolean
}

const Button: React.FC<IButton> = ({ text, onClick, loading = false }) => {
  return (
    <button
      className="min-w-[80px] flex justify-center p-2 text-gray-200 rounded-md bg-slate-600"
      onClick={() => onClick()}
    >
      {loading ? <Loading loading /> : text}
    </button>
  )
}

export default Button
