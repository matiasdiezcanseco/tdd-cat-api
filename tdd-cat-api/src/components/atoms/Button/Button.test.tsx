import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Button from './Button'

describe('test Button', () => {
  let onClick = jest.fn()

  beforeEach(() => {
    onClick = jest.fn()
  })

  test('should display text', () => {
    render(<Button text="Test" onClick={onClick} />)
    expect(screen.getByText(/test/i)).toBeVisible()
  })

  test('should execute callback on click', async () => {
    render(<Button text="Test" onClick={onClick} />)
    expect(onClick).toHaveBeenCalledTimes(0)
    await userEvent.click(screen.getByText(/test/i))
    expect(onClick).toHaveBeenCalledTimes(1)
  })

  test('should show loading', () => {
    render(<Button text="Test" onClick={onClick} loading />)
    expect(screen.getByTestId('loading-testid')).toBeVisible()
  })

  test('should match snapshot', () => {
    const { container } = render(<Button text="Test" onClick={onClick} />)
    expect(container).toMatchSnapshot()
  })
})
