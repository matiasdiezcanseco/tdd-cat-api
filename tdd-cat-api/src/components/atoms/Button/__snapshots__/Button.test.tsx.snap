// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button should match snapshot 1`] = `
<div>
  <button
    class="min-w-[80px] flex justify-center p-2 text-gray-200 rounded-md bg-slate-600"
  >
    Test
  </button>
</div>
`;
